src/cpp/src/vlm_pipeline.cpp

```diff
@@ -447,31 +608,16 @@ DecodedResults VLMPipeline::generate(
             }
         }
     }
-    m_language.set_tensor("inputs_embeds", inputs_embeds);
-    size_t history_len = m_language.get_tensor("attention_mask").get_shape().at(1);
-    m_language.get_tensor("attention_mask").set_shape({1, history_len + inputs_embeds.get_shape()[1]});
-    std::fill_n(m_language.get_tensor("attention_mask").data<int64_t>(), m_language.get_tensor("attention_mask").get_size(), 1);
-    m_language.get_tensor("position_ids").set_shape({1, inputs_embeds.get_shape().at(1)});
-    std::iota(m_language.get_tensor("position_ids").data<int64_t>(), m_language.get_tensor("position_ids").data<int64_t>() + m_language.get_tensor("position_ids").get_size(), history_len);
-    m_language.get_tensor("beam_idx").set_shape({ BATCH_SIZE });
-    m_language.get_tensor("beam_idx").data<int32_t>()[0] = 0;
+    
+    std::shared_ptr<Sampler> sampler = std::make_shared<Sampler>(m_tokenizer);
 
-    m_language.infer();
+    std::vector<SequenceGroup::Ptr> requests;
+    auto attention_size = m_language.get_tensor("attention_mask").get_size();	    // request_id, input_ids, generation_config, block_size, enable_prefix_caching
+    // now we have one prompt as input, so we need one request
+    SequenceGroup::Ptr sequence_group = std::make_shared<SequenceGroup>(0, encoded_input, generation_config, 1, false);
+    sequence_group->set_sequence_group_ptr(sequence_group);
+    requests.push_back(sequence_group);
 
```

Comment:
could you please also drop VLM specific sampling https://github.com/openvinotoolkit/openvino.genai/blob/master/src/cpp/src/vlm_sampling.hpp ?

Reply:
removed